addon/validations/factory.js

```diff
@@ -272,17 +276,36 @@ function getCPDependentKeysFor(attribute, validations) {
 }
 
 /**
+ * Used to retrieve the validation result by calling the validate method on the validator.
+ * If resolve is passed, that means that this validation has been debounced so we pass the
+ * result to the resolve method.
+ * @param  {Validator} validator
+ * @param  {Object} options
+ * @param  {Object} model
+ * @param  {String} attribute
+ * @param  {Function} resolve
+ */
+function getValidationResult(validator, options, model, attribute, resolve) {
```

Comment:
instead of this function, we should likely just do the following, from the calling side.

```
_ => resolve(validator.validate(get(model, attribute), options, model));
```